front/app/modules/free/user_confirmation/services/confirmation.ts

```diff
@@ -0,0 +1,50 @@
+import { API_PATH } from 'containers/App/constants';
+import streams from 'utils/streams';
+
+export const confirmationApiEndpoint = `${API_PATH}/user/confirm`;
+export const resendCodeApiEndpoint = `${API_PATH}/user/resend_code`;
+
+export type IConfirmation = {
+  code?: string | null;
+};
+
+export async function confirm(confirmation: Partial<IConfirmation>) {
+  const bodyData = {
+    confirmation,
+  };
+  try {
+    await streams.add(confirmationApiEndpoint, bodyData);
+
+    streams.fetchAllWith({
+      apiEndpoint: [`${API_PATH}/users/me`],
+      onlyFetchActiveStreams: true,
+    });
+
+    return true;
+  } catch (errors) {
+    throw errors.json.errors;
+  }
+}
+
+export async function resendCode(newEmail?: string | null) {
+  const bodyData = newEmail
+    ? {
+        new_email: newEmail,
+      }
+    : null;
+
+  try {
+    await streams.add(resendCodeApiEndpoint, bodyData);
+
+    if (bodyData?.new_email) {
+      streams.fetchAllWith({
+        apiEndpoint: [`${API_PATH}/users/me`],
+        onlyFetchActiveStreams: true,
```

Comment:
Would not set `onlyFetchActiveStreams`to true, as in some cases (though very unlikely in practice) it might block the `${API_PATH}/users/me` stream from being refetched

Reply:
Also, it might be safer to `await` the streams.fetchAllWith() as well to make sure the refetch completed before `resendCode` is resolved